### PR TITLE
Log refactor

### PR DESCRIPTION
**Pull request by ramsayleung**
1. refactor log level from debug to info and trace in productTypeController
2. capitalize word in the beginning of log message
3. add period in the end of log message
4. make log message  more compact

### CODE REVIEW

File: build_scripts/quality_assurance/config/pmd/rulesets/strings.xml [hunk 1]
```diff
@@ -294,7 +294,7 @@ new StringBuffer() 		//  16
 new StringBuffer(6)		//  6
 new StringBuffer("hello world")  // 11 + 16 = 27
 new StringBuffer('A')	//  chr(A) = 65
-new StringBuffer("A")   //  1 + 16 = 17 
+new StringBuffer("A")   //  1 + 16 = 17
 
 new StringBuilder() 		//  16
 new StringBuilder(6)		//  6
```
Comment (davis999) [inline build_scripts/quality_assurance/config/pmd/rulesets/strings.xml]: recovery this file.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 2]
```diff
@@ -49,11 +54,12 @@
...
   public ProductTypeView createProductType(@RequestBody
                                            @Valid ProductTypeDraft productTypeDraft) {
-    LOG.debug("enter createProductType, draft is:{}", productTypeDraft.toString());
+    LOG.info("Enter. Create productType: {}.", productTypeDraft);
 
     ProductTypeView result = productTypeService.createProductType(productTypeDraft);
 
-    LOG.debug("end createProductType, new product type is:{}", result.toString());
+    LOG.info(LOG_EXIT_PRODUCTTYPEID,result.getId());
+    LOG.trace("ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 3]
```diff
@@ -67,11 +73,11 @@ public ProductTypeView createProductType(@RequestBody
   @DeleteMapping(Router.PRODUCT_TYPE_WITH_ID)
   public void deleteProductTypeById(@PathVariable(value = Router.PRODUCT_TYPE_ID) String id,
                                     @RequestParam @NotNull Integer version) {
-    LOG.debug("enter deleteProductTypeById, id is : {}, version is : {}", id, version);
+    LOG.info("Enter. ProductTypeId: {}, version: {}.", id, version);
 
     productTypeService.deleteProductTypeById(id, version);
 
-    LOG.debug("end deleteProductTypeById, id is : {}, version is : {}", id, version);
+    LOG.info("Exit. ProductTypeId: {}, version: {}.",id,version);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 4]
```diff
@@ -83,11 +89,11 @@ public void deleteProductTypeById(@PathVariable(value = Router.PRODUCT_TYPE_ID)
   @DeleteMapping(Router.PRODUCT_TYPE_WITH_KEY)
   public void deleteProductTypeByKey(@PathVariable(value = Router.PRODUCT_TYPE_KEY) String key,
                                      @RequestParam @NotNull Integer version) {
-    LOG.debug("enter deleteProductTypeById, key is : {}, version is : {}", key, version);
+    LOG.info("Enter. ProductTypeKey: {}, version: {}.",key,version);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 5]
```diff
@@ -83,11 +89,11 @@ public void deleteProductTypeById(@PathVariable(value = Router.PRODUCT_TYPE_ID)
...
 
     productTypeService.deleteProductTypeByKey(key, version);
 
-    LOG.debug("end deleteProductTypeById, key is : {}, version is : {}", key, version);
+    LOG.info("Exit. ProductTypeKey: {}, version: {}.",key,version);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 6]
```diff
@@ -102,14 +108,14 @@ public void deleteProductTypeByKey(@PathVariable(value = Router.PRODUCT_TYPE_KEY
   public ProductTypeView updateProductTypeById(@PathVariable(value = Router.PRODUCT_TYPE_ID)
                                                    String id,
                                                @RequestBody @Valid UpdateRequest updateRequest) {
-    LOG.debug("enter updateProductTypeById,id is {}, update request is {}",
-        id, updateRequest.toString());
+    LOG.info("Enter. ProductTypeId: {}, update request: {}.",id,updateRequest);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 7]
```diff
@@ -102,14 +108,14 @@ public void deleteProductTypeByKey(@PathVariable(value = Router.PRODUCT_TYPE_KEY
...
 
     ProductTypeView result = productTypeService.updateProductTypeById(id, updateRequest
             .getVersion(),
         updateRequest.getActions());
 
-    LOG.debug("end updateCategory, updated Category is {}", result.toString());
+    LOG.info(LOG_EXIT_PRODUCTTYPEID,result.getId());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 8]
```diff
@@ -102,14 +108,14 @@ public void deleteProductTypeByKey(@PathVariable(value = Router.PRODUCT_TYPE_KEY
...
+    LOG.trace("Updated ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 9]
```diff
@@ -125,15 +131,13 @@ public ProductTypeView updateProductTypeById(@PathVariable(value = Router.PRODUC
   public ProductTypeView updateProductTypeByKey(@PathVariable(value = Router.PRODUCT_TYPE_ID)
                                                     String key,
                                                 @RequestBody @Valid UpdateRequest updateRequest) {
-    LOG.debug("enter updateProductTypeById,key is {}, update request is {}",
-        key, updateRequest.toString());
-
+    LOG.info("Enter. ProductTypeKey: {}, update request: {}.",key,updateRequest);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 10]
```diff
@@ -125,15 +131,13 @@ public ProductTypeView updateProductTypeById(@PathVariable(value = Router.PRODUC
...
     ProductTypeView result = productTypeService.updateProductTypeByKey(key, updateRequest
             .getVersion(),
         updateRequest.getActions());
 
-    LOG.debug("end updateCategory, updated Category is {}", result.toString());
-
+    LOG.info(LOG_EXIT_PRODUCTTYPEID,result.getId());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 11]
```diff
@@ -125,15 +131,13 @@ public ProductTypeView updateProductTypeById(@PathVariable(value = Router.PRODUC
...
+    LOG.trace("Updated ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 12]
```diff
@@ -146,9 +150,10 @@ public ProductTypeView updateProductTypeByKey(@PathVariable(value = Router.PRODU
   @GetMapping(Router.PRODUCT_TYPE_WITH_ID)
   public ProductTypeView getProductTypeById(@PathVariable(value = Router.PRODUCT_TYPE_ID) String
                                                 id) {
-    LOG.debug("enter getProductTypeById, id is:{}", id);
+    LOG.info("Enter. ProductTypeId: {}.",id);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 13]
```diff
@@ -146,9 +150,10 @@ public ProductTypeView updateProductTypeByKey(@PathVariable(value = Router.PRODU
...
     ProductTypeView result = productTypeService.getProductTypeById(id);
-    LOG.debug("end getProductTypeById, get ProductType : {}", result.toString());
+    LOG.info(LOG_EXIT_PRODUCTTYPEID,result.getId());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 14]
```diff
@@ -146,9 +150,10 @@ public ProductTypeView updateProductTypeByKey(@PathVariable(value = Router.PRODU
...
+    LOG.trace("ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 15]
```diff
@@ -161,11 +166,11 @@ public ProductTypeView getProductTypeById(@PathVariable(value = Router.PRODUCT_T
   @GetMapping(Router.PRODUCT_TYPE_WITH_KEY)
   public ProductTypeView getProductTypeByKey(@PathVariable(value = Router.PRODUCT_TYPE_KEY)
                                                  String key) {
-    LOG.debug("enter getProductTypeByKey, key is : {}", key);
-
+    LOG.info("Enter. ProductTypeKey: {}.",key);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 16]
```diff
@@ -161,11 +166,11 @@ public ProductTypeView getProductTypeById(@PathVariable(value = Router.PRODUCT_T
...
     ProductTypeView result = productTypeService.getProductTypeByKey(key);
 
-    LOG.debug("end getProductTypeById, get ProductType : {}", result.toString());
+    LOG.info(LOG_EXIT_PRODUCTTYPEID,result.getId());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 17]
```diff
@@ -161,11 +166,11 @@ public ProductTypeView getProductTypeById(@PathVariable(value = Router.PRODUCT_T
...
+    LOG.trace("ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 18]
```diff
@@ -178,12 +183,12 @@ public ProductTypeView getProductTypeByKey(@PathVariable(value = Router.PRODUCT_
    */
   @GetMapping(Router.PRODUCT_TYPE_ROOT)
   public PagedQueryResult<ProductTypeView> queryProductTypes(QueryConditions queryConditions) {
-    LOG.debug("enter queryProductTypes, QueryConditions is : {}", queryConditions.toString());
-
+    LOG.info("Enter. Query parameters: {}.",queryConditions);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 19]
```diff
@@ -178,12 +183,12 @@ public ProductTypeView getProductTypeByKey(@PathVariable(value = Router.PRODUCT_
...
     PagedQueryResult<ProductTypeView> result = productTypeService.queryProductTypes(
                                                                               queryConditions);
 
-    LOG.debug("end queryProductTypes, productType number is : {}", result.getResults().size());
+    LOG.info("Exit. Result count: {}.",result.getResults().size());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 20]
```diff
@@ -178,12 +183,12 @@ public ProductTypeView getProductTypeByKey(@PathVariable(value = Router.PRODUCT_
...
+    LOG.trace("Query result: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 21]
```diff
@@ -32,6 +32,11 @@
   private static final Logger LOG = LoggerFactory.getLogger(ProductTypeService.class);
 
   /**
+   * Log exit with product type id.
+   */
+  private static final String LOG_EXIT_PRODUCTTYPEID="Exit. ProductTypeId: {}.";
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 22]
```diff
@@ -60,7 +65,8 @@ public ProductTypeView createProductType(ProductTypeDraft productTypeDraft) {
 
     ProductTypeView result = ProductTypeMapper.toModel(savedEntity);
 
-    LOG.debug("end createProductType, new product type is:{}", result.toString());
+    LOG.debug("Exit. New productTypeId: {}.", result.getId());
+    LOG.trace("New productType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 23]
```diff
@@ -81,7 +87,8 @@ public void deleteProductTypeById(String id, Integer version) {
     productTypeRepository.delete(id);
     // TODO: 16/12/8 send a message
 
-    LOG.debug("end deleteProductTypeById, id is : {}, version is : {}", id, version);
+    LOG.debug("Exit. Deleted productTypeId: {}, version: {}.", id, version);
+    LOG.trace("Deleted productType: {}.",entity);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 24]
```diff
@@ -102,7 +109,8 @@ public void deleteProductTypeByKey(String key, Integer version) {
 
     // TODO: 16/12/8 send a message
 
-    LOG.debug("end deleteProductTypeById, key is : {}, version is : {}", key, version);
+    LOG.debug("Exit. Deleted productTypeId: {}, version: {}.",entity.getId(),version);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 25]
```diff
@@ -102,7 +109,8 @@ public void deleteProductTypeByKey(String key, Integer version) {
...
+    LOG.trace("Deleted productType: {}.",entity);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 26]
```diff
@@ -115,13 +123,14 @@ public void deleteProductTypeByKey(String key, Integer version) {
...
       actions) {
-    LOG.debug("enter updateProductTypeById, id is : {}, version is : {}, update actions is : {}",
+    LOG.debug("Enter. ProductTypeId: {}, version: {}, update actions: {}.",
         id, version, actions);
 
     ProductType entity = getProductTypeEntityById(id);
     ProductTypeView result = updateProductTypeEntity(version, actions, entity);
 
-    LOG.debug("end updateProductTypeById, updated ProductType is {}", result.toString());
+    LOG.debug(LOG_EXIT_PRODUCTTYPEID,result.getId());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 27]
```diff
@@ -115,13 +123,14 @@ public void deleteProductTypeByKey(String key, Integer version) {
...
+    LOG.trace("Updated ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 28]
```diff
@@ -135,13 +144,14 @@ public ProductTypeView updateProductTypeById(String id, Integer version, List<Up
...
                                                 List<UpdateAction> actions) {
-    LOG.debug("enter updateProductTypeByKey, key is : {}, version is : {}, update actions is : {}",
+    LOG.debug("Enter. ProductTypeKey: {}, version: {}, update actions: {}.",
         key, version, actions);
 
     ProductType entity = getProductTypeEntityByKey(key);
     ProductTypeView result = updateProductTypeEntity(version, actions, entity);
 
-    LOG.debug("end updateProductTypeByKey, updated ProductType is {}", result.toString());
+    LOG.debug(LOG_EXIT_PRODUCTTYPEID,result.getId());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 29]
```diff
@@ -135,13 +144,14 @@ public ProductTypeView updateProductTypeById(String id, Integer version, List<Up
...
+    LOG.trace("Updated productType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 30]
```diff
@@ -152,12 +162,13 @@ public ProductTypeView updateProductTypeByKey(String key, Integer version,
...
   public ProductTypeView getProductTypeById(String id) {
-    LOG.debug("enter getProductTypeById, id is:{}", id);
+    LOG.debug("Enter. ProductTypeId: {}.", id);
 
     ProductType entity = getProductTypeEntityById(id);
     ProductTypeView result = ProductTypeMapper.toModel(entity);
 
-    LOG.debug("end getProductTypeById, get ProductType:{}", result.toString());
+    LOG.debug(LOG_EXIT_PRODUCTTYPEID, result);
+    LOG.trace("ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 31]
```diff
@@ -168,12 +179,13 @@ public ProductTypeView getProductTypeById(String id) {
...
   public ProductTypeView getProductTypeByKey(String key) {
-    LOG.debug("enter getProductTypeByKey, key is : {}", key);
+    LOG.debug("Enter. ProductTypeKey: {}.", key);
 
     ProductType entity = getProductTypeEntityByKey(key);
     ProductTypeView result = ProductTypeMapper.toModel(entity);
 
-    LOG.debug("end getProductTypeByKey, get ProductType : {}", result.toString());
+    LOG.debug(LOG_EXIT_PRODUCTTYPEID, result.getId());
+    LOG.trace("ProductType: {}.",result);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 32]
```diff
@@ -185,16 +197,16 @@ public ProductTypeView getProductTypeByKey(String key) {
    */
   // TODO: 16/12/13 queryconditions
   public PagedQueryResult<ProductTypeView> queryProductTypes(QueryConditions queryConditions) {
-    LOG.debug("enter queryProductTypes, QueryConditions is : {}", queryConditions.toString());
-
+    LOG.debug("Enter. Query conditions: {}.",queryConditions);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 33]
```diff
@@ -185,16 +197,16 @@ public ProductTypeView getProductTypeByKey(String key) {
...
     List<ProductType> entities = productTypeRepository.findAll();
     List<ProductTypeView> productTypes = ProductTypeMapper.toModel(entities);
 
     PagedQueryResult<ProductTypeView> result = new PagedQueryResult<>();
     result.setResults(productTypes);
     result.setTotal(productTypes.size());
 
-    LOG.debug("end queryProductTypes, product number is : {}", productTypes.size());
+    LOG.debug("Exit. Queried productType count: {}.",productTypes.size());
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java [hunk 34]
```diff
@@ -185,16 +197,16 @@ public ProductTypeView getProductTypeByKey(String key) {
...
+    LOG.trace("ProductTypes: {}.",productTypes);
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: need a space.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java [hunk 35]
```diff
@@ -35,6 +35,11 @@
   private static final Logger LOG = LoggerFactory.getLogger(ProductTypeController.class);
 
   /**
+   * Log exit with product type id.
+   */
+  private static final String LOG_EXIT_PRODUCTTYPEID="Exit. ProductTypeId: {}.";
```
Comment (davis999) [inline src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: need space.

File: build_scripts/quality_assurance/config/pmd/rulesets/strings.xml (hunk 1 above)
Comment (ramsayleung) [reply build_scripts/quality_assurance/config/pmd/rulesets/strings.xml]: recoverd.

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 35 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 2 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 3 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 4 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 5 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 6 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 7 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 8 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 9 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 10 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 11 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 12 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 13 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 14 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 15 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 16 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 17 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 18 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 19 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java (hunk 20 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/application/controller/ProductTypeController.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 21 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 22 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 23 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 24 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 25 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 26 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 27 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 28 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 29 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 30 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 31 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 32 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 33 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

File: src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java (hunk 34 above)
Comment (ramsayleung) [reply src/main/java/io/reactivesw/producttype/domain/service/ProductTypeService.java]: added space

**Review by davis999 — APPROVED** [review]

**Pull request merged**